snappy/snap_local.go

```diff
@@ -57,87 +59,115 @@ func NewInstalledSnap(yamlPath string) (*Snap, error) {
 	return snap, nil
 }
 
+func completeInfo(info *snap.Info, manifest *diskManifest) {
+	// XXX: we actually should not trust snap.yaml name!
+
+	info.Developer = SideloadedDeveloper
```

Comment:
It doesn't have to be done now if it makes things more complex for you, but let's use an empty Developer when we don't know, instead of the magic marker.